services/osparc-gateway-server/src/osparc_gateway_server/backend/utils.py

```diff
@@ -364,10 +364,21 @@ async def get_next_empty_node_hostname(
             filters={"service": service["ID"]}
         )
         if not service_tasks:
-            raise NoServiceTasksError(f"service {service} has no tasks attached")
+            continue
         for task in service_tasks:
             if task["Status"]["State"] in ("new", "pending"):
-                raise TaskNotAssignedError(f"task {task} is not assigned to a host yet")
+                # some task is not running yet. that is a bit weird
+                if (
+                    service_constraints := task.get("Spec", {})
+                    .get("Placement", {})
+                    .get("Constraints", [])
+                ):
```

Comment:
I think you can safely remove this if, service_constraints would be a list here.